k2/csrc/array_ops_test.cu

```diff
@@ -1295,4 +1296,31 @@ TEST(OpsTest, Array2IndexTest) {
   }
 }
 
+template <typename T>
+static void Array1SortTest() {
+  std::vector<T> data = {3, 2, 5, 1};
```

Comment:
it's better if we can random tests for this (especially the empty case, I'm not convinced by what mgpu does with empty case, IIRC, there was a bug about this before in `GetTransposeReordering`)

Reply:
Will add tests for random cases and empty cases.

Reply:
Fixed.